cocodump/target_loader.py

```diff
@@ -0,0 +1,34 @@
+import importlib
+from collections.abc import Callable
+from pathlib import Path
+
+from cocodump.base_types import DecodedSection
+
+
+class TargetDecoder:
+    _decode_func: Callable[[bytearray, bool], DecodedSection]
+
+    def __init__(self, func: Callable[[bytearray, bool], DecodedSection]) -> None:
+        self._decode_func = func
+
+    def decode(self, image: bytearray, has_ivt: bool = False) -> DecodedSection:
+        return self._decode_func(image, has_ivt)
+
+
+def import_target_decoder(target_name: str):
+    decoder_func = \
+        importlib.import_module(f"cocodump.targets.{target_name}.decoder").decode
+
+    return TargetDecoder(decoder_func)
```

Comment:
This can be changed to module `typing.Protocol`, the object wrapper here is useless.

Reply:
Yep.